lib/io.toit

```diff
@@ -71,32 +75,44 @@ abstract mixin Writer:
   If the writer can't write all the data at once tries again until all of the data is
     written. This method is blocking.
   */
-  write data/Data from/int=0 to/int=data.byte-size -> none:
+  write data/Data from/int=0 to/int=data.byte-size --flush/bool=false -> none:
     while not is-closed_:
       from += try-write data from to
-      if from >= to: return
+      if from >= to:
+        if flush: this.flush
+        return
       yield
     assert: is-closed_
     throw "WRITER_CLOSED"
 
   /**
   Writes a single byte.
   */
-  write-byte value/int -> none:
+  write-byte value/int --flush/bool=false -> none:
     cache := byte-cache_
     if not cache:
       cache = ByteArray 1
       byte-cache_ = cache
     cache[0] = value
-    write cache
+    write cache --flush=flush
 
   /**
   Writes all data that is provided by the given $reader.
   */
-  write-from reader/Reader -> none:
+  write-from reader/Reader --flush/bool=false -> none:
     if is-closed_: throw "WRITER_CLOSED"
     while data := reader.read:
       write data
+    if flush: this.flush
+
+  /**
+  Flushes any buffered data to the underlying resource.
+
+  Often, one can just use the `--flush` flag of the $write, $write-byte or $write-from
+    functions instead.
+  */
+  flush -> none:
+    flush_
```

Comment:
Why is flush split into two functions?

Reply:
I wanted to have the methods that can be implemented by subclasses next to each other and "protected" (with a trailing "_").

I guess there are two options now:
1. just let people overwrite `flush`.
2. at least call `flush_` from all the other places (instead of `this.flush` in `write-from` ...).

Reply:
I think it is a bit artificial to split it here, but I think both options you provide are sound.